app/models/user.server.ts

```diff
@@ -9,6 +11,30 @@ export async function getUserById(id: User['id']) {
     return prisma.user.findUnique({ where: { id } });
 }
 
+export async function getUserSettings(userId: User['id']) {
+    return prisma.userSettings.findUnique({ where: { userId } });
+}
+
+export async function updateUserColorScheme(request: Request) {
+    // User must be logged in
+    const user = await getUser(request);
+    if (!user) {
+        throw redirect('/');
```

Comment:
nitpick(blocking): The paradigm I have setup here is that these functions do not know or care about remix concepts. This should simply throw and error and whatever thing is calling this `updateUserColorScheme` function will handle and deal with it. This will make it easier to remove or change the frontend framework without a huge refactor. Lots of room to improve it though :D 

Also, throwing something other than a `Error` object is a js smell so let's change to an `Error` anyways

Reply:
Better yet, this function doesn't need to care about checking for the user either. Just pass the `userId` like in the `getUserSettings` function above :)